api/transport/response.go

```diff
@@ -47,3 +51,21 @@ type ResponseWriter interface {
 	// of Write().
 	SetApplicationError()
 }
+
+// ApplicationFailure describes an application failure
+type ApplicationFailure struct {
```

Comment:
We can probably serve the reader well by adding some more information here, e.g.:

```
// ApplicationErrorDetails contains additional information to describe the
// application error, such as an error name and code. This information is
// optional for backwards-compatibility and may not be present in all
// responses.
```